swarm/storage/resource_test.go

```diff
@@ -65,161 +69,102 @@ func (r *FakeRPC) BlockNumber() (string, error) {
 }
 
 // check that signature address matches update signer address
-func TestResourceSignature(t *testing.T) {
+func TestResourceReverse(t *testing.T) {
 
-	// privkey for signing updates
-	privkey, err := crypto.GenerateKey()
-	if err != nil {
-		return
-	}
+	period := uint32(4)
+	version := uint32(2)
 
-	// set up rpc and create resourcehandler
-	rh, _, err, teardownTest := setupTest(privkey, nil, nil)
+	// signer containing private key
+	signer, err := newTestSigner()
 	if err != nil {
-		teardownTest(t, err)
+		t.Fatal(err)
 	}
 
-	// create a new resource
-	validname, err := idna.ToASCII(domainName)
+	// set up rpc and create resourcehandler
+	rh, _, _, teardownTest, err := setupTest(nil, newTestValidator(signer.signContent))
 	if err != nil {
 		teardownTest(t, err)
```

Comment:
Let's defer the tearDownTest at the beginning of the test code, and call Fatal here. That will also guarantee running the tearDown, and it will not hide where the test failed.
In this case you also don't have to call Fatal in the teardown function

Reply:
This is taken care of it `swarm-mutableresources-apinew`